kyuubi-util-scala/src/main/scala/org/apache/kyuubi/util/reflect/ReflectUtils.scala

```diff
@@ -70,20 +78,26 @@ object ReflectUtils {
    * @return
    */
   def invokeAs[T](target: AnyRef, methodName: String, args: (Class[_], AnyRef)*): T = {
-    val targetClass = target.getClass
+    val (clz, obj) = getTargetClass(target)
     val argClasses = args.map(_._1)
     try {
-      DynMethods.builder(methodName)
-        .hiddenImpl(targetClass, argClasses: _*)
-        .buildChecked(target)
-        .invoke[T](args.map(_._2): _*)
+      val method = DynMethods.builder(methodName)
+        .hiddenImpl(clz, argClasses: _*)
+        .impl(clz, argClasses: _*)
+        .buildChecked
+      if (method.isStatic) {
+        method.asStatic.invoke[T](args.map(_._2): _*)
```

Comment:
not sure if it's a good idea to include static case

Reply:
I lean to separate the static case to dedicated method

Reply:
how about the static fields. Separate a method for them as well？

Reply:
After my second thought, a dedicated method for static class may cause confusion for usage, eg. a method of Sacla object class is not static. The usages for both object/static class and common classes are shown in test cases which clear enough to understand and use. I would keep it as it is with auto detecting for whether it's a static one.